### PR TITLE
Hook up editor badges to roles api

### PR DESCRIPTION
**Pull request by terryttsai**
## Description of the change
Screen Shot 2023-01-27 at 9 28 27 AM

Show editor badges in reports and data entry form

Still need to update API to show badges in UploadedFiles page tho!

## Type of change

> All pull requests must have at least one of the following labels applied (otherwise the PR will fail):

| Label                       	| Description                                                                                               	|
|-----------------------------	|-----------------------------------------------------------------------------------------------------------	|
| Type: Bug                   	| non-breaking change that fixes an issue                                                                   	|
| Type: Feature               	| non-breaking change that adds functionality                                                               	|
| Type: Breaking Change       	| fix or feature that would cause existing functionality to not work as expected                            	|
| Type: Non-breaking refactor 	| change addresses some tech debt item or prepares for a later change, but does not change functionality    	|
| Type: Configuration Change  	| adjusts configuration to achieve some end related to functionality, development, performance, or security 	|
| Type: Dependency Upgrade      | upgrades a project dependency - these changes are not included in release notes                             |

## Related issues

Closes #XXXX

### Development

**This box MUST be checked by the submitter prior to merging**:
- [ ] **Double- and triple-checked that there is no Personally Identifiable Information (PII) being mistakenly added in this pull request**

These boxes should be checked by the submitter prior to merging:
- [ ] Tests have been written to cover the code changed/added as part of this pull request

### Code review

These boxes should be checked by reviewers prior to merging:

- [ ] This pull request has a descriptive title and information useful to a reviewer
- [ ] This pull request has been moved out of a Draft state, has no "Work In Progress" label, and has assigned reviewers
- [ ] Potential security implications or infrastructural changes have been considered, if relevant

### CODE REVIEW

**Review by mxosman — APPROVED** [review]
LGTM! Looks like you're adding new props `name` and `editor` and updating permissions. The `mocks/reportOverviews.json` was deceivingly the largest diff.

**Comment by lilidworkin** [comment]
If it's okay let's hold on merging this @terryttsai  until I have a chance to review too!

File: common/types.ts [hunk 1]
```diff
@@ -20,6 +20,12 @@ export enum Permission {
   AGENCY_ADMIN = "agency_admin",
 }
 
+export enum AgencyRole {
```
Comment (lilidworkin) [inline common/types.ts:23]: This overlaps with @ivan-kishko 's PR [here](https://github.com/Recidiviz/justice-counts/pull/347/files) -- might be fine but just letting you guys know to be careful of merge conflicts

File: common/types.ts [hunk 2]
```diff
@@ -90,7 +101,7 @@ export interface ReportOverview {
   last_modified_at: string | null;
   // TODO(#14138): Backend should only send timestamps
   last_modified_at_timestamp: number | null;
-  editors: string[];
+  editors: ReportEditor[];
```
Comment (lilidworkin) [inline common/types.ts:104]: One idea, which I mentioned on Nichelle's PR and tagged you in, is to have the backend send the new format in a new editors_v2 field. That would prevent the situation we got in last week where you had to rush to get this PR in to unbreak main. I don't feel too strongly, the alternative is just to make sure you both merge your PRs at roughly the same time.

**Review by lilidworkin — APPROVED** [review]
Looks good to me! Just coordinate with @nichelle-hall about whether you'll land your PRs at the same time or use an editors_v2 field to make them independent.

**Pull request merged**